src/huggingface_hub/utils/_cache_manager.py

```diff
@@ -577,10 +579,7 @@ def scan_cache_dir(cache_dir: Optional[Union[str, Path]] = None) -> HFCacheInfo:
 
     cache_dir = Path(cache_dir).expanduser().resolve()
     if not cache_dir.exists():
-        raise ValueError(
-            f"Cache directory not found: {cache_dir}. Please use `cache_dir` argument"
-            " or set `HUGGINGFACE_HUB_CACHE` environment variable."
-        )
+        raise CacheNotFound(cache_dir)
```

Comment:
I would still put an explicit message in the error on how to set the cache dir in `scan_cache_dir`. For a Python use, that's more explicit. And if you need to parse the output, it's still possible quite easily. 

```suggestion
        raise CacheNotFound(
            f"Cache directory not found: {cache_dir}. Please use `cache_dir` argument"
            " or set `HUGGINGFACE_HUB_CACHE` environment variable."
        )
```